src/seeding.jl

```diff
@@ -40,22 +40,20 @@ initseeds_by_costs(algname::Symbol, costs::AbstractMatrix{<:Real}, k::Integer) =
 initseeds(iseeds::Vector{Int}, X::AbstractMatrix{<:Real}, k::Integer) = iseeds
 initseeds_by_costs(iseeds::Vector{Int}, costs::AbstractMatrix{<:Real}, k::Integer) = iseeds
 
-function copyseeds!(S::AbstractMatrix{T}, X::AbstractMatrix{T},
-                    iseeds::AbstractVector) where T<:Real
-    d = size(X, 1)
-    n = size(X, 2)
+function copyseeds!(S::Matrix{TS}, X::AbstractMatrix{TX},
+                    iseeds::AbstractVector) where {TS<:AbstractFloat, TX<:Real}
+    d, n = size(X)
     k = length(iseeds)
-    (size(X,1) == d && size(S) == (d, k)) ||
-        throw(DimensionMismatch("Inconsistent array dimensions."))
-
+    size(S) == (d, k) || throw(DimensionMismatch("Inconsistent array " *
+                                                 " dimensions."))
     for j = 1:k
-        copyto!(view(S,:,j), view(X,:,iseeds[j]))
+        copyto!(view(S, :, j), view(X, :, iseeds[j]))
     end
     return S
 end
 
 copyseeds(X::AbstractMatrix{<:Real}, iseeds::AbstractVector) =
-    copyseeds!(similar(X, size(X, 1), length(iseeds)), X, iseeds)
+    copyseeds!(Matrix{eltype(X)}(undef, size(X, 1), length(iseeds)), X, iseeds)
```

Comment:
That makes sense if `X` is `SparseArray`. Is `SparseArray` case covered by any tests?

Reply:
I'll check and will add relevant comment or remove as appropriate

Reply:
What was the problem with `similar`?

Reply:
AFAIK, `similar()` will return a sparse matrix if `X` is sparse. Which is not we want.